flatten_json.go

```diff
@@ -139,6 +146,7 @@ func (fj *flattenJSON) readObject(pathNode SegmentsTreeTracker) error {
 		return err
 	}
 
+	// how many leaf steps (fieldsCount) and chidStructures (nodesCount) have been mentioned in patterns?
 	fieldsCount := pathNode.FieldsCount()
```

Comment:
Maybe we should rename the variables to `leafSteps` and `childStructures` (or `childNodes`) so it will be more clear?

Reply:
You're sounding like me, now. Will have a closer look.

Reply:
Having thought about this, I'm OK. Because if they understand pathSegements at all, they will have figured out what we mean by fields and nodes.